ultralytics/nn/tasks.py

```diff
@@ -80,6 +82,10 @@ def _predict_once(self, x, profile=False, visualize=False):
             y.append(x if m.i in self.save else None)  # save output
             if visualize:
                 feature_visualization(x, m.type, m.i, save_dir=visualize)
+            if embed and m.i in embed:
+                embeddings.append(x)
+                if m.i == embed[-1]:
```

Comment:
probably using max(embed) makes more sense, embed[-1] to break is assuming user using ascending order but users are unpredictable

Reply:
@Laughing-q ah yes, good point. Fixed now!